cl/people_db/lookup_utils.py

```diff
@@ -240,13 +255,114 @@
 IS_JUDGE = {"wu", "re", "du", "de"}
 
 
-def extract_judge_last_name(text: str) -> List[str]:
+def find_all_judges(judge_text: str) -> [str]:
+    """Find all judges
+
+    This method is used to extract out multiple judge names from a text input
+    from the harvard merger/import.
+
+    :param text: Harvard text input from judges tags
+    :return: List of judges names or empty list
+    """
+    cleaned_text = judge_text.replace("\n", " ")[:100].strip()
+    cleaned_text = cleaned_text.replace("By the Court", "")
+    cleaned_text = cleaned_text.replace(" and", ", and")
+    query1 = re.findall(
+        "(((Van|De|Da)\s)?[A-Z][\w\-'']{3,}(\s(IV|I|II|III|V|Jr\.|Sr\.))?),?",
+        cleaned_text,
+    )
+    query2 = re.findall(
+        ",\sand\s(((Van|De|Da)\s)?[A-Z][\w\-'']{3,}(\s(IV|I|II|III|V|Jr\.|Sr\.))?)",
+        cleaned_text,
+    )
+    query = query1 + query2
+    if query:
+        matches = [
+            name[0] for name in query if name[0].lower() not in NOT_JUDGE_WORDS
+        ]
+        return matches
+    return []
+
+
+def find_just_name(text: str) -> str:
+    """Extract the first surname appearing in the Harvard text
+
+    This is designed specifically for the Harvard merger and its particular
+    brand of OCR.
+
+    :param text: string to analyze
+    :return: surname or empty string
+    """
+    # Crop the text - on the off chance the text is incorrectly long to avoid
+    # searching the text way down.
+    cleaned_text = text.replace("\n", " ")[:100].strip()
+    # this is done to handle weird OCR issue
+    cleaned_text = cleaned_text.replace("By the Court", "")
+
+    # First we extract out PER CURIAM
+    if "PER CURIAM" in cleaned_text.upper():
+        return "PER CURIAM"
+
+    # OCR typically fails on the per curiam but this was an easy way to
+    # make sure we recognized it.
+    match_per_curiam = re.search(r"(Pe. C......)", cleaned_text)
+    if match_per_curiam:
+        return "PER CURIAM"
+
+    # The most common style matches here NAME_IN_ALL_CAPS.
```

Comment:
Not quite.

Reply:
This is the style that it is meant to capture...

`ARNOLD, Circuit Judge, with whom BRIGHT, Senior Circuit Judge, and McMILLIAN and MAGILL, Circuit Judges, join`

this is an odd one - because it is both the author and the list of judges.  In this case - we can search for the `ARNOLD` and then compile the rest for all judges on the panel.